src/Uno.UI/UI/Xaml/UIElement.TextHelper.wasm.cs

```diff
@@ -154,10 +154,14 @@ internal void SetForeground(object localValue)
 					WindowManagerInterop.SetElementColor(HtmlId, scb.ColorWithOpacity);
 					break;
 				case GradientBrush gradient:
+					// background-size not supported for inlines (e.g, Run) because DesiredSize is always zero there.
+					// In fact, inlines shouldn't have DesiredSize in the first place as they are TextElement → DependencyObject
+					// They don't inherit from UIElement.
 					this.SetStyle(
 						("background", gradient.ToCssString(this.RenderSize)),
 						("color", "transparent"),
-						("background-clip", "text")
+						("background-clip", "text"),
+						("background-size", this is TextBlock ? $"{this.DesiredSize.Width}px" : "auto")
```

Comment:
I'm not seeing the `background-size` style in the element from the sample:
![image](https://github.com/unoplatform/uno/assets/5839577/a7c190ba-dbd9-495b-a1b5-331a8f48da22)

Do you see it?

Reply:
Or maybe it's automatically folded in background.

Reply:
@jeromelaban The browser embeds it inside the `background` itself. `/ 363px` part.
Same with `background-clip: "text"` which also becomes part of `background`